script.js

```diff
@@ -1,11 +1,18 @@
 // Create an array called `colors` that contains three colors.
-
+var colors = ["red", "blue","green"]
 // Log the last item in the array to the console.
-
+console.log(colors.length - 1)
 // Create a `for` loop that prints out "My favorite color is [insert color here]" for each item in `colors`.
-
+for (var i = 0; i < 3 i ++)
+console.log("My favourite color is" + [i] )
```

Comment:
You're forgetting a few of the key components of a for loop...

- The `for` line is missing a `;` between `i < 3` and `i++`

- After the `for( ... )` you need to open a pair of `{ }`. The content of the loop (i.e., `console.log( ... )`) goes inside of these curly brackets.

- You want to print out whatever color is located at index `i` inside the `colors` array. `[i]` alone will just print out an array with that number in it. You should be using `colors[i]`. Otherwise, your code has no idea that it needs to extract a color from the `colors` array.

In the end, it should look like this...

```js
for (var i = 0; i < 3 i ++){
  console.log("My favourite color is " + colors[i] )
}
```